src/backend/gslice.c

```diff
@@ -0,0 +1,256 @@
+
+        if ((s->Sflags & (GTregcand | SFLunambig)) != (GTregcand | SFLunambig))
+        {
+            sia[si].canSlice = false;
+            continue;
+        }
+
+        targ_size_t sz = type_size(s->Stype);
+        if (sz != 2 * REGSIZE ||
+            tyfv(s->Stype->Tty) || tybasic(s->Stype->Tty) == TYhptr)    // because there is no TYseg
+        {
+            sia[si].canSlice = false;
+            continue;
+        }
+
+        switch (s->Sclass)
+        {
+            case SCfastpar:
+            case SCregister:
+            case SCauto:
+            case SCshadowreg:
+            case SCparameter:
+                anySlice = true;
+                sia[si].canSlice = true;
+                break;
+
+            case SCstack:
+            case SCpseudo:
+            case SCstatic:
+            case SCbprel:
+                sia[si].canSlice = false;
+                break;
+
+            default:
+                symbol_print(s);
+                assert(0);
+        }
+    }
+
+    if (!anySlice)
+        goto Ldone;
+
+    for (block *b = startblock; b; b = b->Bnext)
+    {
+        if (b->BC == BCasm)
+            goto Ldone;
+        if (b->Belem)
+            sliceStructs_Gather(sia, b->Belem);
+    }
+
+    {
+        bool any = false;
+        int n = 0;
+        for (int si = 0; si < sia_length; si++)
+        {
+            sia2[si + n].canSlice = false;
+            if (sia[si].canSlice)
+            {
+                Symbol *sold = globsym.tab[si + n];
+
+                size_t idlen = 2 + strlen(sold->Sident) + 2;
+                char *id = (char *)malloc(idlen + 1);
+                assert(id);
+                sprintf(id, "__%s_%d", sold->Sident, REGSIZE);
+                if (debugc) printf("creating slice symbol %s\n", id);
+                Symbol *snew = symbol_calloc(id, idlen);
+                free(id);
+                snew->Sclass = sold->Sclass;
+                snew->Sfl = sold->Sfl;
+                snew->Sflags = sold->Sflags;
+                if (snew->Sclass == SCfastpar || snew->Sclass == SCshadowreg)
+                {
+                    snew->Spreg = sold->Spreg2;
+                    snew->Spreg2 = NOREG;
+                    sold->Spreg2 = NOREG;
+                }
+                type_free(sold->Stype);
+                sold->Stype = type_fake(TYnptr);
+                sold->Stype->Tcount++;
+                snew->Stype = type_fake(TYnptr);
+                snew->Stype->Tcount++;
+
+                SYMIDX sinew = symbol_add(snew);
+                for (int i = sinew; i > si + n + 1; --i)
+                {
+                    globsym.tab[i] = globsym.tab[i - 1];
+                    globsym.tab[i]->Ssymnum += 1;
+                }
+                globsym.tab[si + n + 1] = snew;
+                snew->Ssymnum = si + n + 1;
+
+                sia2[si + n].canSlice = true;
+                sia2[si + n].si0 = si + n;
+                ++n;
+                any = true;
+            }
```

Comment:
This body is complex, very low-level, does sth. unexpected (seems like it's replacing the symbol in the global symtab), and comes without explanation/comment. Please add at least a short sentence for the reasoning (what and why), instead of having readers spent 5min. to unpuzzle the low level code, and 10min. on understanding.